lib/src/header/all_day_events.dart

```diff
@@ -42,15 +44,25 @@ class AllDayEvents<E extends Event> extends StatelessWidget {
               events = events.where((e) => e.intersectsInterval(visibleDates));
 
               return ValueListenableBuilder(
-                valueListenable: controller.scrollControllers.pageListenable,
-                builder: (context, page, __) =>
-                    _buildEventLayout(context, events, page),
-              );
+                  valueListenable: controller.scrollControllers.pageListenable,
+                  builder: (context, page, __) => GestureDetector(
+                        behavior: HitTestBehavior.translucent,
+                        onTapUp: (details) {
+                          final tappedCell = details.localPosition.dx / (constraints.widthConstraints().maxWidth / controller.visibleRange.visibleDays);
+                          final date = LocalDate.fromEpochDay(page.floor() + tappedCell.toInt());
```

Comment:
When we're currently viewing page `8.7` (e.g., while scrolling) and tap cell `1.6`, this would calculate day `8 + 1 = 9`, whereas we actually tapped day `8.7 + 1.6 = 10.3 ≈ 10`. This can be fixed by first adding both `double`'s and only then `floor()` the result